subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/DefaultPrecompiledScriptPluginsSupport.kt

```diff
@@ -246,7 +246,7 @@ fun Project.enableScriptCompilationOf(
         kotlinCompileTask {
             dependsOn(generatePrecompiledScriptPluginAccessors)
             inputs.files(compileClasspath).withNormalizer(ClasspathNormalizer::class.java)
-            inputs.dir(accessorsMetadata).withPathSensitivity(PathSensitivity.RELATIVE)
+            inputs.dir(accessorsMetadata).withPathSensitivity(PathSensitivity.RELATIVE).ignoreEmptyDirectories()
```

Comment:
While you are at it, could you add names as well to the these inputs added via the runtime API?

Reply:
yes!